Fix ACK.Terminal state after controller upgrade

knottnt opened the pull request:
Issue #, if available: [2699](https://github.com/aws-controllers-k8s/community/issues/2699)

Description of changes:
After an upgrade of the emrcontainers-controller the the delta detected in `Spec.Tags` due to the change in controller version results in and ACK.Terminal state due to `sdkUpdate` not being implemented. To avoid this behavior this PR adds support for updating tags to `VirtualCluster`. Due to some additional complexity in tag support this PR simply removes `Spec.Tags` from the delta comparison for `JobRun` to avoid the change in this field from triggering an update. 

- Add support for updating tags to VirtualCluster to allow updating of tags when controller version delta detected
- Ignore JobRun Spec.Tags in delta comparison to avoid attempted update of controller version tag following upgrade of controller version

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

knottnt committed:
Fix ACK.Terminal state after controller upgrade

- Add support for updating tags to VirtualCluster to allow updating of tags when controller version delta detected

- Ignore JobRun Spec.Tags in delta comparison to avoid attempted update of controller version tag following upgrade of controller version

michaelhtm commented on generator.yaml:
```diff
@@ -73,6 +75,9 @@ resources:
           is_ignored: true
+      Tags:
+        compare:
+          is_ignored: true
```
if you ignore tag comparison there will never be a delta detected here..

knottnt replied:
Marked JobRun's Spec.Tags as immutable to allow users to get feedback when an explicit changes is made to the field. There will be a change in behavior due to this. Namely, instead of a `not implemented` Terminal condition the user will now see the change rejected by the kubernetes api server.

michaelhtm reviewed (commented):
Thanks @knottnt 
left a few comments below

knottnt committed:
Mark JobRun Spec.Tags as immutable

- Set JobRun Spec.Tags as immutable to allow for user feedback when an explicit change is made to the field.

michaelhtm reviewed (approved):
/lgtm
